### PR TITLE
Find & Display Terms

### PR DESCRIPTION
**Pull request by ninaprabhu**

### CODE REVIEW

**Comment by Tetheta** [comment]
Starting Review

File: src/main/java/com/google/sps/servlets/TermServlet.java [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+  };
+
+  final String SEMESTER_TERMS[] = {
+    "Spring", "Spring", "Spring", "Spring", "Spring", "Summer",
+    "Summer", "Fall", "Fall", "Fall", "Fall", "Fall"
+  };
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    Boolean quarter = isQuarter(request.getParameter("school-name"));
```
Comment (Tetheta) [inline src/main/java/com/google/sps/servlets/TermServlet.java:49]: Another point where I'd name the boolean isQuarter and the function getIsQuarter or something like that.

File: src/main/java/com/google/sps/servlets/TermServlet.java [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+      season = QTR_TERMS[month];
+    } else {
+      season = SEMESTER_TERMS[month];
+    }
+    int year = cal.get(Calendar.YEAR);
+    return new Term(season + " " + String.valueOf(year), isQuarter);
+  }
+
+  /* Go num terms forwards and backwards. */
+  private List<Term> getTerms(Term currTerm, int num) {
```
Comment (Tetheta) [inline src/main/java/com/google/sps/servlets/TermServlet.java:81]: num isn't a very descriptive variable name here. is this numTerms? Maybe call it maxTermOffset or something?

File: src/main/java/com/google/sps/servlets/TermServlet.java [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+    "Summer", "Fall", "Fall", "Fall", "Fall", "Fall"
+  };
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    Boolean quarter = isQuarter(request.getParameter("school-name"));
+    JSONObject json = new JSONObject();
+    json.put("quarter", quarter);
+    Term currTerm = getCurrTerm(quarter);
+    List<String> terms = termsToString(getTerms(currTerm, 2));
```
Comment (Tetheta) [inline src/main/java/com/google/sps/servlets/TermServlet.java:53]: The 2 here would be good to annotate with /* variable name= */

File: src/main/webapp/login-script.js [hunk 4]
```diff
@@ -4,10 +4,19 @@ async function signIn(googleUser) {
...
   url.searchParams.set("private", "false");
-  const response = await fetch(url, { method: "POST" });
-  const id = await response.json();
+  const loginResponse = await fetch(url, { method: "POST" });
+  const id = await loginResponse.json();
   if (id.verified) {
     // Successful sign-in.
+    const termList = await getTermList();
+    const select = document.getElementById("search-term");
+    const opt = document.createElement('option');
```
Comment (Tetheta) [inline src/main/webapp/login-script.js:13]: I'd call this optionElement, little clearer that it's not optional (which is usually what opt means)

File: src/main/webapp/login-script.js [hunk 5]
```diff
@@ -67,3 +77,39 @@ function signOutPrivate() {
...
+  const start = email.indexOf("@");
+  const end = email.lastIndexOf(".");
+  return email.substring(start + 1, end);
+}
+
+function createOptionElement(val) {
+  const opt = document.createElement('option');
+  opt.appendChild(document.createTextNode(val));
+  opt.value = val; 
+  return opt;
```
Comment (Tetheta) [inline src/main/webapp/login-script.js:103]: Better naming again for val and opt, though not as critical here as the function name is good.

File: src/main/webapp/login-script.js [hunk 6]
```diff
@@ -67,3 +77,39 @@ function signOutPrivate() {
...
+}
+
+function removeAllTerms() {
+  parent = document.getElementById("search-term");
+  while (parent.firstChild) {
+    parent.removeChild(parent.firstChild);
+  }
+  const opt = document.createElement('option');
```
Comment (Tetheta) [inline src/main/webapp/login-script.js:111]: Another nit on opt naming

File: src/main/java/com/google/sps/servlets/TermServlet.java [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+
+  final String QTR_TERMS[] = {
+    "Winter", "Winter", "Winter", "Spring", "Spring", "Spring",
+    "Summer", "Summer", "Summer", "Fall", "Fall", "Fall"
+  };
+
+  final String SEMESTER_TERMS[] = {
+    "Spring", "Spring", "Spring", "Spring", "Spring", "Summer",
+    "Summer", "Fall", "Fall", "Fall", "Fall", "Fall"
+  };
```
Comment (Tetheta) [inline src/main/java/com/google/sps/servlets/TermServlet.java:45]: Nifty way to do this. Simple but effective.

Comment (ninaprabhu) [reply src/main/java/com/google/sps/servlets/TermServlet.java:45]: Thank you! :D

**Pull request merged**